recipes/openldap/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+    license = "OLDAP-2.8"
+    topics = ("ldap", "load-balancer", "directory-access")
+    exports_sources = ["patches/*"]
+    settings = settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cyrus_sasl": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cyrus_sasl": True
+
+    }
+    _autotools = None
+    _configure_vars = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def requirements(self):
+        self.requires("openssl/1.1.1m")
+        if self.options.with_cyrus_sasl:
+            self.requires("cyrus-sasl/2.1.27")
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                f"{self.name} is only supported on Linux")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+
+        def yes_no(v): return "yes" if v else "no"
+        self._autotools = AutoToolsBuildEnvironment(self)
+        configure_args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+            "--with-cyrus_sasl={}".format(yes_no(self.options.with_cyrus_sasl)),
+            "--with-pic={}".format(yes_no(self.options.get_safe("fPIC", True))),
+            "--without-fetch",
+            "--with-tls=openssl",
+            "--enable-auditlog"]
+        self._configure_vars = self._autotools.vars
+        self._configure_vars["systemdsystemunitdir"] = os.path.join(
+            self.package_folder, "res")
+
+        # Need LD_LIBRARY_PATH for configure. Otherwise, configure fails.
+        libpath = ""
+        for path in self._autotools.library_paths:
+            libpath += "{}:".format(path)
+        self._configure_vars["LD_LIBRARY_PATH"] = libpath
+
+        # Need to link to -pthread instead of -lpthread for gcc 8 shared=True
+        # on CI job. Otherwise, linking fails.
+        self._autotools.libs.remove("pthread")
+        self._configure_vars["LIBS"] = self._configure_vars["LIBS"].replace(
+            "-lpthread", "-pthread")
+
+        self._autotools.configure(
+            args=configure_args,
+            configure_dir=self._source_subfolder,
+            vars=self._configure_vars)
+        return self._autotools
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        autotools = self._configure_autotools()
+
+        autotools.make(args=["depend"], vars=self._configure_vars)
```

Comment:
```suggestion
```

Not needed, only more CI time wasted. All dependencies are already solved, don't need to scan to check it